### PR TITLE
Installer v2

### LINKED ISSUE #3785: [FR] Add interacive options to installer
### Please verify that this feature request has NOT been suggested before.

- [X] I checked and didn't find similar feature request

### Problem statement

The installer currently is non-interactive. It can be modified with environment variables but it would be cool to have an interactive guide to pick a database backend and maybe set up Redis ... there are options.

### Suggested solution

Create an interactive installer:
- [ ] 'no input' switch (default on)
- [ ] database selection (sqlite, postgres, MySQL)
- [ ] add cache (redis)
- [ ] (optional) create optional installer pingback

### Describe alternatives you've considered

The new installer in #3743 is a great improvement (IMO, I wrote it) but could be more interactive.

### Examples of other systems

The installer for [Coolify](https://docs.coollabs.io/coolify/installation) is great, something like it could be used as a blueprint.

### Do you want to develop this?

- [X] I want to develop this.

### LINKED ISSUE #3835: [FR] Add warning for unsupported OSes
### Please verify that this feature request has NOT been suggested before.

- [X] I checked and didn't find similar feature request

### Problem statement

Ubuntu 22.04 is not supported right now, this leads to confusion if the docs are not read (the one-liner is shared in multiple places).
See https://github.com/inventree/InvenTree/issues/3825

### Suggested solution

Add a check and warning for not supported OSs and versions

### Describe alternatives you've considered

RTFD is always an option but
a) not really in the spirit of the project
b) we then should remove the one-liner everywhere and point to the docs

### Examples of other systems

_No response_

### Do you want to develop this?

- [X] I want to develop this.

### PR DESCRIPTION
**Pull request by matmair**
WIP
This PR changes the installer to a sh script that should be run interactively. Details:
- Adds CLI helper
- Select source (official inventree or fork, stable or master)

Missing (in addition to #3785):
- [x] docs

Contains things for #3785
Closes #3835

<a href="https://gitpod.io/#https://github.com/inventree/InvenTree/pull/3798"></a>

### CODE REVIEW

**Comment by matmair** [comment]
@SchrodingersGat I would like to squeeze this into the 0.9.0 release. Still needs some work but it would enable easier modifications of the installer and maybe also warnings for Ubuntu 22.04, CentOS etc.

**Comment by matmair** [comment]
@SchrodingersGat the warning is now in; I would appreciate a check for Debian, ubuntu (20.04 and 22.04) and maybe SEL; I used the current DO images to do the checks but I want to be sure this time

**Comment by matmair** [comment]
Ok, featurewise this is as far as I would like it to be in 0.9.0 - all the fancy stuff will come in a later PR.

File: contrib/installer/src/root_command.sh [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+function do_call() {
+    if [[ $dry_run ]]; then
+        echo -e "### DRY RUN: \n$1"
+    else
+        $1
+    fi
+}
+
+function get_distribution {
```
Comment (SchrodingersGat) [inline contrib/installer/src/root_command.sh:18]: @matmair I see this is a duplicate of `contrib/installer/install` - not sure if there is a way around this thought? LMK what you think

Comment (matmair) [reply contrib/installer/src/root_command.sh:18]: @SchrodingersGat this is intended. `contrib/installer/install` is generated by combining the root command and all the boilerplate to get a nice CLI interface, selectable options etc. Then there is a symlink between `contrib/installer/install` and `contrib/install.sh` to keep the path as short as possible (and backwards compatible).

There is a link to the used [tool](https://bashly.dannyb.co/installation/) in the README `contrib/installer/README`

Comment (SchrodingersGat) [reply contrib/installer/src/root_command.sh:18]: @matmair fair enough, I'm happy with this then

**Pull request merged**